my-sms/spec/controllers/students_controller_spec.rb

```diff
@@ -23,119 +25,123 @@
 # removed from Rails core in Rails 5, but can be added back in via the
 # `rails-controller-testing` gem.
 
-RSpec.describe StudentsController, :type => :controller do
-
+RSpec.describe StudentsController, type: :controller do
   # This should return the minimal set of attributes required to create a valid
   # Student. As you add validations to Student, be sure to
   # adjust the attributes here as well.
-  let(:valid_attributes) {
-    skip("Add a hash of attributes valid for your model")
-  }
-
-  let(:invalid_attributes) {
-    skip("Add a hash of attributes invalid for your model")
-  }
 
   # This should return the minimal set of values that should be in the session
   # in order to pass any filters (e.g. authentication) defined in
   # StudentsController. Be sure to keep this updated too.
   let(:valid_session) { {} }
 
-  describe "GET #index" do
-    it "returns a success response" do
-      student = Student.create! valid_attributes
-      get :index, {}, valid_session
-      expect(response).to be_success
+  describe 'GET' do
+    subject(:student) { FactoryBot.create(:student) }
+
+    context '#index' do
+      it 'returns a success response' do
+        get :index, {}, valid_session
+        expect(response).to be_success
```

Comment:
Here I would put the student in a let binding, and wrap the `get :index` in a subject.

```suggestion
    let(:student) { FactoryBot.create(:student) }
    subject { get :index, {}, valid_session }

    context '#index' do
      it 'returns a success response' do
        expect(subject).to be_success
```

You can then take this one step further by using `shoulda` matchers instead of `expectation` matchers, like:

```ruby
describe "GET /student/:id" do

  # Subject is always the same in this `describe` block
  subject { get :show, { id: student.id } }

  context "with valid student" do

    # with each context we can change the value of `student`
    # which will in turn update our request since it uses the `student.id`
    let(:student) { create :student }
    
    it "does the right thing" do
      # using the shoulda matchers implicitly calls the subject so
      # there is no need to manually call it
      should be_success
      should render_template('student/show')
      # etc..
    end
  end
end
```

Reply:
I've added this in 6cccd5ae169559b61c06f9a81151c62838717fdc